File: backend/app/schemas/me.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MyQuestionOut(BaseModel):
    id: uuid.UUID
    author_id: uuid.UUID
    title: str
    body: str
    category: str
    stage: str
    accepted_answer_id: uuid.UUID | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class MyAnswerOut(BaseModel):
    id: uuid.UUID
    question_id: uuid.UUID
    author_id: uuid.UUID
    body: str
    is_accepted: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/me.ts
export interface MyQuestion {
  id: string;
  author_id: string;
  title: string;
  body: string;
  category: string;
  stage: string;
  accepted_answer_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface MyAnswer {
  id: string;
  question_id: string;
  author_id: string;
  body: string;
  is_accepted: boolean;
  created_at: string;
  updated_at: string;
}
